mucommander-core/src/main/java/com/mucommander/ui/main/WindowManager.java

```diff
@@ -260,13 +281,25 @@ private void setLookAndFeel(String lnfName) {
             ClassLoader oldLoader = currentThread.getContextClassLoader();
             currentThread.setContextClassLoader(ExtensionManager.getClassLoader());
 
-            UIManager.setLookAndFeel(lnfName);
+            try {
+                var currentLaF = UIManager.getLookAndFeel();
+                if (currentLaF == null || !lnfName.equals(currentLaF.getClass().getName())) {
+                    UIManager.setLookAndFeel(lnfName);
+                }
+            } catch (UnsupportedLookAndFeelException e) {
+                LOGGER.info("The chosen Look and feel (that is non-standard) must be loaded now...");
+                // TODO - load the required LaF and defer the rest?
+                loadAdditionalLookAndFeels(); // we defer that to that point as it takes some time :)
+                UIManager.setLookAndFeel(lnfName);
+            }
 
             // Restores the contextual ClassLoader.
             currentThread.setContextClassLoader(oldLoader);
 
             mainFrames.forEach(MainFrame::updateFileTablesHeaderRenderer);
-            mainFrames.forEach(SwingUtilities::updateComponentTreeUI);
+            mainFrames.forEach(e -> {
+                SwingUtilities.updateComponentTreeUI(e.getJFrame());
+            });
```

Comment:
how about mapping each MainFrame to JFrame and then proceed with the forEach?

Reply:
later in other PR? (it's kinda overwhelming me already, especially after a couple of months from initial commit :) )